deps/v8/src/sampler.cc

```diff
@@ -342,6 +342,15 @@ void SignalHandler::HandleProfilerSignal(int signal, siginfo_t* info,
     // We require a fully initialized and entered isolate.
     return;
   }
+
+  // ISSUE-14576: To avoid deadlock, return if there is a thread lock
+  if (Isolate::GetProcessWideMutex().Pointer()->TryLock()) {
+    Isolate::GetProcessWideMutex().Pointer()->Unlock();
+  }
```

Comment:
Style: I'm not familiar with V8's code base's style guidelines, but I think this line should be `} else {` (curly braces on the same line as the `else` keyword).